djangooidc/views.py

```diff
@@ -141,41 +188,46 @@ def logout(request, next_page=None):
-        url += "?" + urlencode(request_args)
-        return HttpResponseRedirect(url)
-
-        # Looks like they are implementing back channel logout, without checking for
-        # support?
-        # http://openid.net/specs/openid-connect-backchannel-1_0.html#Backchannel
-        """
-        request_args = None
-        if 'id_token' in request.session.keys():
-            request_args = {'id_token': IdToken(**request.session['id_token'])}
-        res = client.do_end_session_request(state=request.session["state"],
-                                            extra_args=extra_args, request_args=request_args)
-        content_type = res.headers.get("content-type", "text/html") # In case the logout response doesn't set content-type (Seen with Keycloak)
-        resp = HttpResponse(content_type=content_type, status=res.status_code, content=res._content)
-        for key, val in res.headers.items():
-            resp[key] = val
-        return resp
-        """
+        if 'access_token' in request.session and 'state' in request.session:
+            # TODO: determine if OIDC was used to login by some better way
+
+            request_args = {
+                'id_token_hint': request.session['access_token'],
+                'state': request.session['state'],
+            }
+            # should include the post_logout_redirect_uri
+            request_args.update(extra_args)
+
+            url = client.provider_info['end_session_endpoint']
+            url += "?" + urlencode(request_args)
+            return HttpResponseRedirect(url)
+
+            # Looks like they are implementing back channel logout, without checking for
+            # support?
+            # http://openid.net/specs/openid-connect-backchannel-1_0.html#Backchannel
+            """
+            request_args = None
+            if 'id_token' in request.session.keys():
+                request_args = {'id_token': IdToken(**request.session['id_token'])}
+            res = client.do_end_session_request(state=request.session["state"],
+                                                extra_args=extra_args, request_args=request_args)
+            content_type = res.headers.get("content-type", "text/html") # In case the logout response doesn't set content-type (Seen with Keycloak)
+            resp = HttpResponse(content_type=content_type, status=res.status_code, content=res._content)
+            for key, val in res.headers.items():
+                resp[key] = val
+            return resp
+            """
     finally:
-        # Always remove Django session stuff - even if not logged out from OP. Don't wait for the callback as it may never come.
+        # Always remove Django session stuff - even if not logged out from OP.
+        # Don't wait for the callback as it may never come.
         auth_logout(request)
         if next_page:
             request.session['next'] = next_page
 
+    return HttpResponseRedirect(next_page)
```

Comment:
next_page may be None
Should probably redirect to '/' so similar if variable is not set